### PR TITLE
Add option `tlsIntercept`

### PR DESCRIPTION
**Pull request by lipsumar**
This PR adds an option `tlsIntercept` to `MockttpHttpsOptions`. This option is the inverse of the existing `tlsPassthrough`:
- `tlsPassthrough`: intercept everything except hostnames in this list
- `tlsIntercept`: intercept nothing except hostnames in this list

Both options can not be used at the same time.

This option allows `mockttp` to be less intrusive and to let most requests untouched, except for a specific list of hostnames. This helps keep some applications using certificate pinning working correctly.

Related issue: #162 

@pimterry let me know if you'd prefer more tests and/or if docs need to be adjusted.

### CODE REVIEW

**Comment by lipsumar** [comment]
@pimterry sorry for the delay on signing the CLA, I needed approval. We're all set!

**Review by pimterry — APPROVED** [review]
Thanks for that @lipsumar. The changes all look good here. Unfortunately though the CI tests have highlighted that the current test suite fails in the latest version of Node that was just released (v20.12.0) due to some kind of HTTP/2 changes, so I'm going to need to get that resolved before merging.

I'll look into that shortly and let you know once it's sorted so we can rebase this & merge it happily.

**Comment by pimterry** [comment]
Thanks @lipsumar! Rebased & merged :+1:. I've renamed the option slightly here and gone for `tlsInterceptOnly` to make it a bit clearer what this is doing.

I'll release this shortly as part of v3.11.0.

**Comment by lipsumar** [comment]
`tlsInterceptOnly` makes a lot of sense, good call 👍 Thanks for your reactivity !

**Pull request merged**
